core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -83,6 +87,10 @@ protected SerializableTable(Table table) {
     this.encryption = table.encryption();
     this.locationProvider = table.locationProvider();
     this.refs = SerializableMap.copyOf(table.refs());
+    this.metricsReporterProperties =
+        table instanceof BaseTable
+            ? SerializableMap.copyOf(table.metricsReporter().properties())
+            : null;
```

Comment:
we should rather do https://github.com/apache/iceberg/pull/7144#issuecomment-1504987582 than setting this to null here.

Reply:
Sorry, I missed the comment when I try to fix the CI, I wll change it later.